src/fi_tostr.c

```diff
@@ -803,8 +797,31 @@ char *DEFAULT_SYMVER_PRE(fi_tostr)(const void *data, enum fi_type datatype)
 		break;
 	}
 	return buf;
+
+}
+
+__attribute__((visibility ("default"),EXTERNALLY_VISIBLE))
+char *DEFAULT_SYMVER_PRE(fi_tostr)(const void *data, enum fi_type datatype)
+{
+	static char *buf = NULL;
+
+	if (!buf) {
+		buf = calloc(OFI_BUFSIZ, 1);
+		if (!buf)
+			return NULL;
+	}
+
+	return ofi_tostr(data, datatype, buf, OFI_BUFSIZ);
 }
 DEFAULT_SYMVER(fi_tostr_, fi_tostr, FABRIC_1.0);
 
+__attribute__((visibility ("default"),EXTERNALLY_VISIBLE))
+char *DEFAULT_SYMVER_PRE(fi_tostr_r)(const void *data, enum fi_type datatype,
+		char *buf, size_t buflen)
+{
+	return ofi_tostr(data, datatype, buf, buflen);
+}
+DEFAULT_SYMVER(fi_tostr_r_, fi_tostr_r, FABRIC_1.0);
```

Comment:
This should be CURRENT_SYMVER.  Please see include/ofi_abi.h for a description of handling ABI changes.  I need to figure out if we bump the current ABI or not as a result.  I don't think so, but I can't remember.